File: dotnet/Core/OpenStack/Platform_System.cs
using OpenStack.Sfx;
using System;

namespace OpenStack;

/// <summary>
/// SystemAudioBuilder
/// </summary>
public class SystemAudioBuilder : AudioBuilderBase<object> {
    public override object CreateAudio(object path) => throw new NotImplementedException();
    public override void DeleteAudio(object audio) => throw new NotImplementedException();
}

#region OpenSfx

/// <summary>
/// SystemSfx
/// </summary>
public class SystemSfx(ISource source) : IOpenSfx<object> {
    readonly ISource _source = source;
    readonly AudioManager<object> _audioManager = new(source, new SystemAudioBuilder());

    public ISource Source => _source;
    public AudioManager<object> AudioManager => _audioManager;
    public object CreateAudio(object path) => _audioManager.CreateAudio(path).aud;
}

#endregion

File: Python/openstk/platforms/system/platform_system.py
from __future__ import annotations
import os, io, pathlib
from openstk.core import ISource, BinaryReader, PlatformX
from openstk.sfx import IOpenSfx2, AudioBuilderBase, AudioManager

#region Platform

# SystemAudioBuilder
class SystemAudioBuilder(AudioBuilderBase):
    def createAudio(self, path: object) -> object: raise NotImplementedError()
    def deleteAudio(self, audio: object) -> None: raise NotImplementedError()

# SystemSfx
class SystemSfx(IOpenSfx2):
    def __init__(self, source: ISource):
        self.source: ISource = source
        self.audioManager: AudioManager = AudioManager(source, SystemAudioBuilder())
    def createAudio(self, path: object) -> int: return self.audioManager.createAudio(path)[0]

#endregion
